TablesContext could not extract schema name for openGauss CREATE INDEX SQL

## Bug Report

### Which version of ShardingSphere did you use?

master branch

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

JDBC

### Expected behavior

Get correct schema name from parsed CREATE INDEX SQL.

### Actual behavior

Run PostgreSQLMigrationGeneralE2EIT on openGauss, it failed.
I added a new unit test `OpenGaussSQLParserEngineTest`, it could not get correct schema name too.

### Reason analyze (If you can)

`TablesContext` doesn't extract owner as schema name:
<img width="2816" height="1310" alt="Image" src="https://github.com/user-attachments/assets/6e306163-26c2-43a8-ad7a-b68ede799fb2" />

--
Some test on openGauss:
```
openGauss=# select * from ds_0.schema1.t_user;
ERROR:  cross-database references are not implemented: "ds_0.schema1.t_user"
LINE 1: select * from ds_0.schema1.t_user;
                      ^
openGauss=# \c ds_0
Non-SSL connection (SSL connection is recommended when requiring high-security)
You are now connected to database "ds_0" as user "omm".

ds_0=# select * from schema1.t_user;
 id | username  | password | schema1_flag | email  
----+-----------+----------+--------------+--------
```

`cross-database references are not implemented`, so could we just use owner as schema name for openGauss?

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

Run unit test on `OpenGaussSQLParserEngineTest` (need to change assert code near TODO comment).

### Example codes for reproduce this issue (such as a github link).

See `OpenGaussSQLParserEngineTest`.

Test on openGauss, run PostgreSQLMigrationGeneralE2EIT .

Proxy log:
```
[INFO ] 2026-01-22 11:46:27.980 [j0102p000033170cdbde8047c808003a086ae4ca0a_Worker-1] o.a.s.d.p.c.p.d.PipelineJobDataSourcePreparer - Execute target table SQL: CREATE INDEX idx_user_id ON test.test.t_order USING btree (user_id) TABLESPACE pg_default
[ERROR] 2026-01-22 11:46:28.225 [j0102p000033170cdbde8047c808003a086ae4ca0a_Worker-1] o.a.s.d.p.c.j.e.DistributedPipelineJobExecutor - Job j0102p000033170cdbde8047c808003a086ae4ca0a-0 execution failed.
org.apache.shardingsphere.database.exception.core.exception.syntax.sql.DialectSQLParsingException: You have an error in your SQL syntax: CREATE INDEX idx_user_id ON test.test.t_order USING btree (user_id) TABLESPACE pg_default, null
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.getException(ShardingSphereSQLParserEngine.java:66)
	at org.apache.shardingsphere.infra.parser.ShardingSphereSQLParserEngine.parse(ShardingSphereSQLParserEngine.java:59)
	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.createQueryContext(ShardingSphereStatement.java:259)
	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.execute(ShardingSphereStatement.java:247)
	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.execute(ShardingSphereStatement.java:195)
	at org.apache.shardingsphere.data.pipeline.core.preparer.datasource.PipelineJobDataSourcePreparer.executeTargetTableSQL(PipelineJobDataSourcePreparer.java:152)
	at org.apache.shardingsphere.data.pipeline.core.preparer.datasource.PipelineJobDataSourcePreparer.prepareTargetTables(PipelineJobDataSourcePreparer.java:124)
	at org.apache.shardingsphere.data.pipeline.scenario.migration.preparer.MigrationJobPreparer.prepareTarget(MigrationJobPreparer.java:148)
```

Original SQL:
```
CREATE INDEX idx_user_id ON test.t_order USING btree (user_id) TABLESPACE pg_default
```
Related debug screenshots:
<img width="3178" height="1362" alt="Image" src="https://github.com/user-attachments/assets/f2befa3c-698f-4872-8a91-c277d300c1ca" />

--- a/kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/metadata/generator/PipelineDDLDecorator.java
+++ b/kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/metadata/generator/PipelineDDLDecorator.java
@@ -25,11 +25,13 @@ import org.apache.shardingsphere.infra.binder.context.statement.SQLStatementCont
 import org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData;
 import org.apache.shardingsphere.infra.metadata.database.schema.util.IndexMetaDataUtils;
 import org.apache.shardingsphere.infra.parser.SQLParserEngine;
+import org.apache.shardingsphere.sql.parser.statement.core.extractor.TableExtractor;
 import org.apache.shardingsphere.sql.parser.statement.core.segment.SQLSegment;
 import org.apache.shardingsphere.sql.parser.statement.core.segment.ddl.constraint.ConstraintSegment;
 import org.apache.shardingsphere.sql.parser.statement.core.segment.ddl.index.IndexSegment;
 import org.apache.shardingsphere.sql.parser.statement.core.segment.generic.table.SimpleTableSegment;
 import org.apache.shardingsphere.sql.parser.statement.core.segment.generic.table.TableNameSegment;
+import org.apache.shardingsphere.sql.parser.statement.core.statement.SQLStatement;
 import org.apache.shardingsphere.sql.parser.statement.core.statement.attribute.SQLStatementAttributes;
 import org.apache.shardingsphere.sql.parser.statement.core.statement.attribute.type.ConstraintSQLStatementAttribute;
 import org.apache.shardingsphere.sql.parser.statement.core.statement.attribute.type.IndexSQLStatementAttribute;
@@ -74,7 +76,7 @@ public final class PipelineDDLDecorator {
         String result = decorateActualSQL(targetDatabaseName, targetTableName, parserEngine, sql.trim());
         // TODO remove it after set search_path is supported.
         if ("openGauss".equals(databaseType.getType())) {
-            return decorateOpenGauss(targetDatabaseName, schemaName, result, parserEngine);
+            return decorateOpenGauss(schemaName, result, parserEngine);
         }
         return Optional.of(result);
     }
@@ -145,27 +147,30 @@ public final class PipelineDDLDecorator {
     }
     
     // TODO remove it after set search_path is supported.
-    private Optional<String> decorateOpenGauss(final String databaseName, final String schemaName, final String queryContext, final SQLParserEngine parserEngine) {
+    private Optional<String> decorateOpenGauss(final String schemaName, final String queryContext, final SQLParserEngine parserEngine) {
         if (queryContext.toLowerCase().startsWith(SET_SEARCH_PATH_PREFIX)) {
             return Optional.empty();
         }
-        return Optional.of(replaceTableNameWithPrefix(queryContext, schemaName, databaseName, parserEngine));
+        return Optional.of(replaceTableNameWithPrefix(queryContext, schemaName, parserEngine));
     }
     
-    private String replaceTableNameWithPrefix(final String sql, final String schemaName, final String databaseName, final SQLParserEngine parserEngine) {
-        SQLStatementContext sqlStatementContext = parseSQL(databaseName, parserEngine, sql);
-        if (sqlStatementContext.getSqlStatement() instanceof CreateTableStatement || sqlStatementContext.getSqlStatement() instanceof CreateIndexStatement
-                || sqlStatementContext.getSqlStatement() instanceof AlterTableStatement || sqlStatementContext.getSqlStatement() instanceof CommentStatement) {
-            if (sqlStatementContext.getTablesContext().getSimpleTables().isEmpty()) {
+    private String replaceTableNameWithPrefix(final String sql, final String schemaName, final SQLParserEngine parserEngine) {
+        SQLStatement sqlStatement = parserEngine.parse(sql, true);
+        if (sqlStatement instanceof CreateTableStatement || sqlStatement instanceof CreateIndexStatement
+                || sqlStatement instanceof AlterTableStatement || sqlStatement instanceof CommentStatement) {
+            TableExtractor tableExtractor = new TableExtractor();
+            tableExtractor.extractTablesFromSQLStatement(sqlStatement);
+            Optional<SimpleTableSegment> simpleTableSegment = tableExtractor.getTableContext().stream()
+                    .filter(each -> each instanceof SimpleTableSegment).map(each -> (SimpleTableSegment) each).findFirst();
+            if (!simpleTableSegment.isPresent()) {
                 return sql;
             }
-            Optional<String> sqlSchemaName = sqlStatementContext.getTablesContext().getSchemaName();
+            Optional<String> sqlSchemaName = simpleTableSegment.get().getOwner().map(optional -> optional.getIdentifier().getValue());
             if (sqlSchemaName.isPresent() && sqlSchemaName.get().equals(schemaName)) {
                 return sql;
             }
             Map<SQLSegment, String> replaceMap = new TreeMap<>(Comparator.comparing(SQLSegment::getStartIndex));
-            TableNameSegment tableNameSegment = sqlStatementContext.getTablesContext().getSimpleTables().iterator().next().getTableName();
-            replaceMap.put(tableNameSegment, schemaName + "." + tableNameSegment.getIdentifier().getValue());
+            replaceMap.put(simpleTableSegment.get(), schemaName + "." + simpleTableSegment.get().getTableName().getIdentifier().getValue());
             return doDecorateActualTable(replaceMap, sql);
         }
         return sql;

